### PR TITLE
Remove randomUUID for 100x speedup

### PR DESCRIPTION
**Pull request by jean-philippe-martin**
A simple change with a big impact.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java [hunk 1]
```diff
@@ -21,12 +23,15 @@
...
+    private final static long uuidHighWord = new Random().nextLong();
+    private final static AtomicLong uuidLowWord = new AtomicLong(0);
 
     private final SAMRecord samRecord;
     private final UUID uuid;
 
     public SAMRecordToGATKReadAdapter( final SAMRecord samRecord ) {
-        this(samRecord, UUID.randomUUID());
+        // this is 100x faster than UUID.randomUUID()
+        this(samRecord, new UUID(uuidHighWord, uuidLowWord.incrementAndGet()));
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java]: Can you please make the equivalent change in the `GoogleGenomicsReadToGATKReadAdapter` for the sake of consistency?

**Comment by droazen** [comment]
:+1: One comment to address then merge at will once tests pass.

File: src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java (hunk 1 above)
Comment (jean-philippe-martin) [inline src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java]: Sure thing!

**Comment by jean-philippe-martin** [comment]
Done, will merge once tests show green.

File: src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java (hunk 1 above)
Comment (laserson) [inline src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java]: And the soon-to-merge `BDGAlignment` version

Comment (laserson) [inline src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java]: Added this in #908

**Pull request merged**